app/src/main/java/io/pumpkinz/pumpkinreader/service/DataSource.java

```diff
@@ -71,6 +73,8 @@ public Comment call(Throwable throwable) {
                         return getInnerComments(comment);
                     }
                 })
+                //TODO: use value from setting
+                .timeout(10, TimeUnit.SECONDS)
```

Comment:
Using the same logic as HTTP request timeout, I don't think we need to make the timeout user-configureable.